internal/transform/transform.go

```diff
@@ -23,67 +23,79 @@ func Transform(doc *tycho.Node, opts TransformOptions) *tycho.Node {
 	shouldScope := len(doc.Styles) > 0 && ScopeStyle(doc.Styles, opts)
 	walk(doc, func(n *tycho.Node) {
 		ExtractScript(doc, n)
+		AddComponentProps(doc, n)
 		if shouldScope {
 			ScopeElement(n, opts)
 		}
 	})
+
+	// Important! Remove scripts from original location *after* walking the doc
+	for _, script := range doc.Scripts {
+		script.Parent.RemoveChild(script)
+	}
+
```

Comment:
This was the fix for scripts

Reply:
Wow this parser is slick! Love the flexibility of this.